### PR TITLE
Add documentation for federated config map

### PR DESCRIPTION
**Pull request by mwielgus**
Same as #1825 but for 1.5 release.

cc: @nikhiljindal @madhusudancs

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1838)

### CODE REVIEW

**Comment by madhusudancs** [comment]
@mwielgus this is cool! Thanks for addressing all the comments. I skimmed through. I hope I did not miss anything.

**Comment by madhusudancs** [comment]
cc @kubernetes/docs

File: docs/user-guide/federation/configmap.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+---
+
+This guide explains how to use ConfigMaps in Federation control plane.
```
Comment (devin-donnelly) [inline docs/user-guide/federation/configmap.md]: Missing an article.

"in a Federation control plane" or "with your Federation control plane."

File: docs/user-guide/federation/configmap.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running Kubernetes Cluster
+Federation installation. If not, then head over to the
+[federation admin guide](/docs/admin/federation/) to learn how to
+bring up a cluster federation (or have your cluster administrator do
+this for you). Other tutorials, for example
+[this one](https://github.com/kelseyhightower/kubernetes-cluster-federation)
```
Comment (devin-donnelly) [inline docs/user-guide/federation/configmap.md]: Make the link title more descriptive than "this one"; think of it like a variable name.

"Other tutorials, such as Kelsey Hightower's [Federated Kubernetes Tutorial](https://github.com/kelseyhightower/kubernetes-cluster-federation), might also help you create a Federated Kubernetes cluster."

File: docs/user-guide/federation/configmap.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+by Kelsey Hightower, are also available to help you.
+
+You are also expected to have a basic
```
Comment (devin-donnelly) [inline docs/user-guide/federation/configmap.md]: "You are also expected to have" -> "You should also have"

File: docs/user-guide/federation/configmap.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+[working knowledge of Kubernetes](/docs/getting-started-guides/) in
+general and [ConfigMaps](/docs/user-guide/ConfigMaps/) in particular.
+
+## Overview
+
+ConfigMaps in federation control plane (referred to as "Federated ConfigMaps" in
```
Comment (devin-donnelly) [inline docs/user-guide/federation/configmap.md]: Just use "Federated ConfigMaps".

"Federated ConfigMaps are very similar to..."

**Comment by devin-donnelly** [comment]
Just a handful of small comments. Thanks!

File: docs/user-guide/federation/configmap.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+this guide) are very similar to the traditional [Kubernetes
+ConfigMaps](/docs/user-guide/configmap/) providing the same functionality.
```
Comment (devin-donnelly) [inline docs/user-guide/federation/configmap.md]: "providing" -> "and provide"

**Comment by devin-donnelly** [comment]
@madhusudancs , @mwielgus , question: Are the new Federated controllers being added to the existing Federation TOC anywhere?

One of these PRs needs to do that, so that users can find the new docs.

**Comment by madhusudancs** [comment]
> @madhusudancs , @mwielgus , question: Are the new Federated controllers being added to the existing Federation TOC anywhere?

@devin-donnelly thanks for reminding that! We haven't done/are not doing that. We probably should. I can send a separate PR for all of these combined.

Also, the TOC for federated resources is not that discoverable. Despite having worked on these things, I had trouble finding them. We should probably make them more discoverable.

**Comment by devin-donnelly** [comment]
@madhusudancs I agree--they were completely missing in 1.4, and I had to add them on very short notice on launch day last time.

I'll get with sig-docs and the other TWs here and see where Federation fits into the new docs organization.

**Comment by madhusudancs** [comment]
@devin-donnelly sure, thanks!

Also, sent a PR for the links - https://github.com/kubernetes/kubernetes.github.io/pull/1848

File: docs/user-guide/federation/configmap.md (hunk 2 above)
Comment (mwielgus) [inline docs/user-guide/federation/configmap.md]: Done.

File: docs/user-guide/federation/configmap.md (hunk 5 above)
Comment (mwielgus) [inline docs/user-guide/federation/configmap.md]: Done.

**Pull request merged**